nautobot/docs/user-guides/custom-fields.md

```diff
@@ -0,0 +1,344 @@
+# Custom Fields
+
+Custom fields are a method of adding new fields to existing Nautobot models. For more general information on how custom fields work, refer to the [custom fields model documentation](../models/extras/customfield.md).
+
+## When to use Custom Fields
+
+Custom fields are commonly used for fields that need different values across individual objects. For example, a custom field on devices to reference an internal ticket number that inventories each device in Nautobot. If you need a solution for marking multiple objects with a common flag, [tags](../models/extras/tag.md) may be a better fit.
+
+## Create a Custom Field
+
+Navigate to the custom fields page by clicking on **Extensibility -> Custom Fields** in the Nautobot menu. Click on **Add** to create a new custom field.
+
+!!! note
+    Custom fields are initialized on associated objects when a content type is added to the custom field, including when the custom field is created. The initial value will be set to the `default` value of the custom field.
+
+### Custom Field Attributes
+
+#### Label
+
+The label is the human readable label of the custom field that will be displayed on the associated object detail view.
+
+![Custom Field Labels](./images/custom-fields/custom_field_detail_label.png)
+
+#### Grouping
+
+The optional grouping field allows you to group custom fields into collapsible menus.
+
+![Custom Field Grouping](./images/custom-fields/custom_field_detail_grouped.png)
+
+#### Slug
+
+The slug is used to create the URL endpoint for the custom field and is also used as the key in the custom field dictionary. This is automatically created from the label if not supplied. The default value should be sufficient for most deployments.
+
```

Comment:
Perhaps add an admonition about using underscores rather than dashes or whitespace if defining a custom slug, similar to the tip that we have in `customfield.md`?